src/translate-adapter.ts

```diff
@@ -0,0 +1,455 @@
+}
+
+async function handleAllCommand(): Promise<void> {
+	await handleTranslateCommand();
+	await handleToWordsCommand();
+	await handleToJsonCommand();
+}
+
+/****************************** Implementation ********************************/
+
+async function translateIoPackage(): Promise<void> {
+	const content = await readJson(ioPackage);
+	if (content.common.news) {
+		console.log("Translate News");
+		for (const k in content.common.news) {
+			console.log(`News: ${k}`);
+			const nw = content.common.news[k];
+			await translateNotExisting(nw);
+		}
+	}
+	if (content.common.titleLang) {
+		console.log("Translate Title");
+		await translateNotExisting(
+			content.common.titleLang,
+			content.common.title,
+		);
+	}
+	if (content.common.desc) {
+		console.log("Translate Description");
+		await translateNotExisting(content.common.desc);
+	}
+	await writeJson(ioPackage, content, { spaces: 4, EOL });
+	console.log(`Successfully updated ${path.relative(".", ioPackage)}`);
+}
+
+async function translateNotExisting(
+	obj: Partial<Record<Language, string>>,
+	baseText?: string,
+): Promise<void> {
+	let text = obj.en;
+	if (!text) {
+		text = baseText;
+	}
+
+	if (text) {
+		for (const lang of getLanguages()) {
+			if (!obj[lang]) {
+				const time = new Date().getTime();
+				obj[lang] = await translateText(text, lang);
+				console.log(
+					gray(`en -> ${lang} ${new Date().getTime() - time} ms`),
+				);
+			}
+		}
+	}
+}
+
+async function translateI18n(baseFile: string): Promise<void> {
+	const filePattern = createFilePattern(baseFile);
+	const baseContent = await readJson(baseFile);
+	const missingLanguages = getLanguages();
+	const files = await findAllLanguageFiles(baseFile);
+	for (const file of files) {
+		const match = file.match(filePattern);
+		const lang = match![2] as Language; // language files always match
+		const langIndex = missingLanguages.indexOf(lang);
+		missingLanguages.splice(langIndex, 1);
+		if (lang === "en") continue;
+		const translation = await readJson(file);
+		await translateI18nJson(translation, lang, baseContent);
+		await writeJson(file, translation, { spaces: 4, EOL });
+		console.log(`Successfully updated ${path.relative(".", file)}`);
+	}
+	for (const lang of missingLanguages) {
+		const translation: Record<string, string> = {};
+		await translateI18nJson(translation, lang, baseContent);
+		const filename = baseFile.replace(filePattern, `$1${lang}$3`);
+		await ensureDir(path.dirname(filename));
+		await writeJson(filename, translation, {
+			spaces: 4,
+			EOL,
+		});
+		console.log(`Successfully created ${path.relative(".", filename)}`);
+	}
+}
+
+async function translateI18nJson(
+	content: Record<string, string>,
+	lang: Language,
+	baseContent: Readonly<Record<string, string>>,
+): Promise<void> {
+	if (lang === "en") {
+		return;
+	}
+	const time = new Date().getTime();
+	for (const t in baseContent) {
+		if (!content[t]) {
+			content[t] = await translateText(baseContent[t], lang);
+		}
+	}
```

Comment:
I really don't like `in` to iterate through objects.
```suggestion
	for (const [t, base] of Object.entries(baseContent)) {
		if (!content[t]) {
			content[t] = await translateText(base, lang);
		}
	}
```